Classes/Issues/Merge/IssueMergeSectionController.swift

```diff
@@ -65,13 +65,28 @@ MergeButtonDelegate {
         var viewModels = [ListDiffable]()
 
         if object.contexts.count > 0 {
-            let success = object.contexts.reduce(true, { $0 && $1.state == .success })
-            viewModels.append(IssueMergeSummaryModel(
-                title: success ?
-                    NSLocalizedString("All checks passed", comment: "")
-                    : NSLocalizedString("Some checks failed", comment: ""),
-                state: success ? .success : .failure
-            ))
+            let states = object.contexts.map { $0.state }
+            let state: IssueMergeSummaryModel.State
+            let stateDescription: String
+            let failureDescription = NSLocalizedString("Some checks failed", comment: "")
+            let containsFailure = states.contains(.failure) || states.contains(.error)
+            switch states {
```

Comment:
Can this entire thing become a `switch true`? Looks like we're not using `states`?

Reply:
Don't think so; the `case let states where ...` uses it.

Reply:
Ah, what I meant is we're not really using `states` here. It's available out of the scope of the `switch` statement. Every case has a `where` clause. So this means what we really care about is a `Bool` and not `states`.
```Swift
switch true {
case containsFailure: //
case states.contains(.pending): //
case states.reduce(true, { $0 && $1 == .success }): //
default: //
}
```
Sorry to be "that guy" 😓

Reply:
Oh wow, I didn't know about that! That is... interesting, to say the least. That would kind of be more readable, huh? 😮

Reply:
Yeah, I do think it is more clear in its intent. I just love `switch true` 🤓

Reply:
Can someone explain why using `switch true {...}` would be better than just `if` statements here?

Reply:
Other than aesthetics, I can’t.

Reply:
No reason other than I prefer how `switch` looks. Ironically, did some digging last night into `switch` vs `if-else` and there is some talk that the compiler can better optimize for a switch. However, even if true, this would be a micro-optimization that has _no real value_ in production. 

Other than that, the switch is capable of using `fallthrough`, which we don't need here. So it is all a matter of preference.